Marlin/src/gcode/motion/G2_G3.cpp

```diff
@@ -35,10 +35,6 @@
   #include "../../module/scara.h"
```

Comment:
Although on its face this commit moving feedrate scaling into the planner would seem to be a great idea, since it simplifies and consolidates this logic, it causes the `inv_segment_length` and `inverse_secs` to be re-calculated on every segment, whereas previously these were only calculated once, at the start of the line segmentation loop. And there's another bug, which I'll point out below…

Reply:
I hadn't thought about losing the optimization of pre-computing `fr_mm_s / millimeters` for every segment by moving the feedrate scaling into the planner. If it's a concern, one option could be to pass it as an optional argument to `buffer_line` in the same way as `millimeters`. Alternatively, `buffer_line` could keep track of the previous `fr_mm_s`, `millimeters`, and the scaled feedrate, and only re-compute the scaled feedrate if `fr_mm_s` or `millimeters` changes.

Reply:
I thought about that. Passing an extra `float` parameter on every segment is definitely going to be less costly than having a division operation. Another option which I've been contemplating would be to get #8311 done, and then possibly the high-level feedrate variables would already contain the duration of the move instead of the speed of the move.